python/gtsam/tests/test_Triangulation.py

```diff
@@ -15,66 +15,108 @@
-        z2 = camera2.project(landmark)
+        z1 = camera1.project(self.landmark)
+        z2 = camera2.project(self.landmark)
 
         # twoPoses
         poses = Pose3Vector()
         measurements = Point2Vector()
 
-        poses.append(pose1)
-        poses.append(pose2)
+        poses.append(self.pose1)
+        poses.append(self.pose2)
         measurements.append(z1)
         measurements.append(z2)
 
         optimize = True
         rank_tol = 1e-9
 
         triangulated_landmark = triangulatePoint3(poses,sharedCal, measurements, rank_tol, optimize)
-        self.gtsamAssertEquals(landmark, triangulated_landmark,1e-9)
+        self.gtsamAssertEquals(self.landmark, triangulated_landmark, 1e-9)
 
         # 2. Add some noise and try again: result should be ~ (4.995, 0.499167, 1.19814)
         measurements = Point2Vector()
         measurements.append(z1 - np.array([0.1, 0.5]))
         measurements.append(z2 - np.array([-0.2, 0.3]))
 
         triangulated_landmark = triangulatePoint3(poses,sharedCal, measurements, rank_tol, optimize)
-        self.gtsamAssertEquals(landmark, triangulated_landmark,1e-2)
-        #
-        # # two Poses with Bundler Calibration
-        # bundlerCal = Cal3Bundler(1500, 0, 0, 640, 480)
-        # camera1 = PinholeCameraCal3Bundler(pose1, bundlerCal)
-        # camera2 = PinholeCameraCal3Bundler(pose2, bundlerCal)
-        #
-        # z1 = camera1.project(landmark)
-        # z2 = camera2.project(landmark)
-        #
-        # measurements = Point2Vector()
-        # measurements.append(z1)
-        # measurements.append(z2)
-        #
-        # triangulated_landmark = triangulatePoint3(poses,bundlerCal, measurements, rank_tol, optimize)
-        # self.gtsamAssertEquals(landmark, triangulated_landmark,1e-9)
+        self.gtsamAssertEquals(self.landmark, triangulated_landmark,1e-2)
+
+    def test_distinct_Ks(self):
+        # two cameras
+        K1 = Cal3_S2(1500, 1200, 0, 640, 480)
+        camera1 = PinholeCameraCal3_S2(self.pose1, K1)
+
+        K2 = Cal3_S2(1600, 1300, 0, 650, 440)
+        camera2 = PinholeCameraCal3_S2(self.pose2, K2)
+
+        cameras = CameraSetCal3_S2()
+        cameras.push_back(camera1)
+        cameras.push_back(camera2)
+
+        # Project two landmarks into two cameras and triangulate
+        z1 = camera1.project(self.landmark)
+        z2 = camera2.project(self.landmark)
+        
+        measurements = Point2Vector()
+        measurements.append(z1)
+        measurements.append(z2)
+
+        optimize = True
+        rank_tol = 1e-9
+
+        triangulated_landmark = triangulatePoint3(cameras, measurements, rank_tol, optimize)
+        self.gtsamAssertEquals(self.landmark, triangulated_landmark, 1e-9)
+
+    def test_distinct_Ks_Bundler(self):
+        # two cameras
+        K1 = Cal3Bundler(1500, 0, 0, 640, 480)
+        camera1 = PinholeCameraCal3Bundler(self.pose1, K1)
+
+        K2 = Cal3Bundler(1600, 0, 0, 650, 440)
```

Comment:
K1, K2, camera1, camera2 can be instantiated in the `setUp` method since they are all common.
Similarly you can generate the measurements in the `setUp` since they all use the same cameras and the same landmark, this will require you to have less copy-paste everywhere.